src/platforms/wayland/wayland_display.cpp

```diff
@@ -48,11 +48,13 @@ void mpw::add_connection_options(boost::program_options::options_description& co
 
 auto mpw::connection(options::Option const& options) -> struct wl_display*
 {
-    if (!options.is_set(wayland_host_option_name))
-        return nullptr;
-
-    static auto const wayland_display =
-        std::make_unique<WaylandDisplay>(options.get<std::string>(wayland_host_option_name).c_str());
+    static auto const wayland_display = std::make_unique<WaylandDisplay>(options.is_set(wayland_host_option_name) ?
+        options.get<std::string>(wayland_host_option_name).c_str() : nullptr);
 
     return wayland_display->wl_display;
 }
+
+auto mir::platform::wayland::supported(mir::options::Option const& options) -> bool
```

Comment:
This is a weirdly-named function now; it's not returning whether mpw is *supported*, but whether it *should* be used. Given it's only used in one place, I think this could be just inlined into `probe_graphics_platform` with a suitable comment.

Reply:
Inlining would require access to (or duplication) of the option name. Simpler to use a better name.